phenotrex/ml/clf/svm.py

```diff
@@ -139,10 +140,19 @@ def get_shap(
             self.logger.error('Cannot create shap values: no Shap explainer trained.')
             return None
         if n_samples is None:
-            n_samples=SHAP_NSAMPLE_DEFAULT
-        self.logger.info(f'Computing SHAP values for input using n_samples={n_samples}.'
-                         f' This may take a long time.')
+            n_samples = SHAP_NSAMPLE_DEFAULT
+        if isinstance(n_samples, str) and n_samples.isnumeric():
+            n_samples = int(n_samples)
+        self.logger.info(f'Computing SHAP values for input using n_samples={n_samples}.')
         raw_feats = self._get_raw_features(records).astype(int)  # numpy error if using bools
-        shap_values = self.shap_explainer.shap_values(raw_feats, nsamples=n_samples)[0]
-        shap_bias = self.shap_explainer.expected_value[0]
+        if raw_feats.shape[1] > SHAP_TRACTABLE_N_FEATURES:
+            too_expensive = f"Attempting to compute SHAP explanation with KernelExplainer and " \
+                            f"n_features={raw_feats.shape[1]}. This may take a _very_ long time."
+            self.logger.warning(too_expensive)
+        shap_values = self.shap_explainer.shap_values(
+            raw_feats,
+            nsamples=n_samples,
+            # TODO: find good value for l1_reg
+        )[1]
+        shap_bias = self.shap_explainer.expected_value[1]
```

Comment:
I think the following is clearer, esp. when the function call spans over several lines:
```suggestion
        _, shap_values = self.shap_explainer.shap_values(
            raw_feats,
            nsamples=n_samples,
            # TODO: find good value for l1_reg
        )
        _, shap_bias = self.shap_explainer.expected_value
```
That is, if the returned object is a 2-tuple. Is it?